src/writing-your-own-application/alloc.md

```diff
@@ -0,0 +1,53 @@
+# Alloc
+
+In `no_std`, we need to rely on `alloc` crate, which provides heap allocation support. To use `alloc` in `no_std` environment:
+
+1. Explicitly enable the alloc crate
+2. Provide a Global Allocator
+
+We provide our own simple `no_std` heap allocator [esp-alloc]. To use it you need to:
+
+1. Add a dependency to your `Cargo.toml`
+```toml
+esp-alloc        = "0.7.0"
+```
+
+2. Modify the `.cargo/config.toml`
+```toml
+[unstable]
+build-std = ["alloc", "core"] # added alloc here
+```
+
+3. Use `alloc` crate in your application
+```rust
+// main.rs
+
+extern crate alloc;
+```
+
+4. Initialize a global heap allocator providing a heap of the given size in bytes (with macro, preferrable)
```

Comment:
```suggestion
4. Initialize a global heap allocator providing a heap of the given size in bytes (with the provided macro, preferably)
```
Would be good to justify this recommendation; why should the macro be preferred?